sound/soc/sof/core.c

```diff
@@ -255,23 +255,22 @@ static int sof_machine_check(struct snd_sof_dev *sdev)
 	return 0;
 }
 
-static int sof_probe(struct platform_device *pdev)
+int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 {
-	struct snd_sof_pdata *plat_data = dev_get_platdata(&pdev->dev);
 	struct snd_sof_dev *sdev;
 	const char *drv_name;
 	const void *mach;
 	int size;
 	int ret;
 
-	sdev = devm_kzalloc(&pdev->dev, sizeof(*sdev), GFP_KERNEL);
+	sdev = devm_kzalloc(dev, sizeof(*sdev), GFP_KERNEL);
 	if (!sdev)
 		return -ENOMEM;
 
-	dev_dbg(&pdev->dev, "probing SOF DSP device....\n");
+	dev_dbg(dev, "probing SOF DSP device....\n");
 
 	/* initialize sof device */
-	sdev->dev = &pdev->dev;
+	sdev->dev = dev;
 	sdev->parent = plat_data->dev;
 	if dev_is_pci(plat_data->dev)
 		sdev->pci = to_pci_dev(plat_data->dev);
```

Comment:
@plbossart this is not utterly important but do we need this "pci" member in snd_sof_dev ? the dev member is already pointing to the top-level device which in case of PCI will be the pci device, so when needed we can always use to_pci_dev(sdev->dev).